### DISCUSSION TITLE
Navigation Menu Draft

### DISCUSSION THREAD

**Discussion by UX-P** [bisq-network/bisq#6092] [Show and tell]
We previously discussed the navigation adjustment and overall, the feedback seemed positive however, a usability test is still important to collect real insight from the appropriate target audience. In the meantime, I have created a quick Figma prototype for anyone to test out and provide feedback/suggestions.

[Lo-fi Navigation Prototype
](https://www.figma.com/proto/OgiN9YXlf8WSqkbz8Rejo3/Menu-Navigation?scaling=min-zoom&page-id=0%3A1&starting-point-node-id=111%3A669&node-id=111%3A669)

**Things to consider:**

If possible, think as a new user and complete various user journeys e.g.

- Buy XMR using Altcoin/Altcoin instant

- Sell/ buy BTC for US dollar/ EUR

- Buy other Altcoin using Altcoin instant

**Bisq contributor/ Stakeholder**

- Which currencies should be displayed?

- Language: e.g. BTC to XMR

- Any technical considerations?

- Thoughts on time-box exercise as suggested by @chimp1984 

@jmacxx @ripcurlx @chimp1984 @m52go

**Comment by chimp1984** [comment]
Thanks @UX-P Looks great!

A few comments:
- Altcoin instant is only used for XMR, so we could omit it for other altcoins. For BSQ the payment methods are: BSQ-Swap and Altcoin.
- The "BTC to Fiat" wording sounds basically good to me but as it introduces a change to current Bisq wording I am wondering if its better if we use "Sell BTC for Fiat" /  "Buy BTC with Fiat" / "Sell XMR for BTC" / "Buy XMR with BTC"? I think that is more in line with current Bisq and should leave no doubt about what is the base and quote currency. But I am fine with your suggestions as well.
- To use "Other" as term for other altcoins/assets is a good choice IMO. Should be enough and is less effort than let user define the tabs.

**Comment by UX-P** [comment]
> "Sell XMR for BTC" / "Buy XMR with BTC"? I think that is more in line with current Bisq and should leave no doubt about what is the base and quote currency.

I agree, and potentially more user friendly as it also replicates real world terms/phrases that could be easily understood by the onboarding and general bisq trader.

**Comment by ripcurlx** [comment]
I think that this navigation hierarchy improves the understanding for new and existing users. Putting everything behind `other` besides  BTC, BSQ and XMR is fine with me.

**Comment by pazza83** [comment]
Thanks for creating this.

I think the two currency dropdowns on the main screen are confusing. Maybe there could just be one drop down at the top (like present) and then the currency can be displayed eg:

- BTC to GBP
- BTC from GBP

I like the Buy / Sell menus but think it might be good to include Fiat as a tab.

Maybe a line of text similar to above.

Buy BTC with ...... [Fiat/BSQ/XMR/Other]
Sell BTC for...... [Fiat/BSQ/XMR/Other]

Maybe default is all markets?

Also the 'Choose payment method' box should filer based on the currency selected. It does not do this in the current Bisq but would be better if it did.

**Reply by ripcurlx** [reply]
> Maybe a line of text similar to above.
> 
> Buy BTC with ...... [Fiat/BSQ/XMR/Other] Sell BTC for...... [Fiat/BSQ/XMR/Other]

But wouldn't that be what we have right now anyways? Besides putting the selection from the dropdown into the tab?

The reason what I like about this new way of navigation is, that it reflects the intention of the buyer/seller.

e.g. I want to buy BTC with SEPA

Buy > BTC > Currency EUR + Payment method SEPA

e.g. I want to sell XMR for BTC

Sell > XMR > Altcoin Instant

Or did you mean something else?

**Reply by pazza83** [reply]
> > Maybe a line of text similar to above.
> > Buy BTC with ...... [Fiat/BSQ/XMR/Other] Sell BTC for...... [Fiat/BSQ/XMR/Other]
> 
> But wouldn't that be what we have right now anyways? Besides putting the selection from the dropdown into the tab?

![Capture](https://user-images.githubusercontent.com/66367547/158493429-8031825b-cb33-47f9-8f14-35386ec48a4f.PNG)

I think I was just meaning `Fiat` is missing from the tabs choices. And that there could be a line of text under the tab that displays the users intention once they have made their selection.

> The reason what I like about this new way of navigation is, that it reflects the intention of the buyer/seller.

Maybe add another slection...

Buy BTC with ...... [Fiat/BSQ/XMR/Other] using [Any/Payment Method]

eg

- Buy BTC with GBP using Faster Payments
- Buy BTC with EUR using SEPA Instant
- Sell BTC for USD using Zelle

**Reply by chimp1984** [reply]
Fiat is not a base currency and would not fit to the BUY/SELL terminology. E.g. We don't say "I buy EUR with BTC".
An alternative would be to put the markets in the submenu. For BTC it would be the dropdown to select BTC-USD, BTC-EUR,... for XMR and BSQ it would be only XMR-BTC and BSQ-BTC. For Other it would be a dropdown as well with all the other altcoins, e.g. ETH-BTC, ZEC-BTC,.... But having 2 dropdowns and 2 buttons for selecting XMR and BSQ would be a bit weird....

**Comment by ripcurlx** [comment]
I do get @m52go's concerns about the navigational hierarchy, putting XMR or another coin on the same hierarchical level as BTC.

So how about we do as suggested, but just move XMR also into the `Other` section.
Having Buy > BTC | BSQ | Other

Doing like that we wouldn't take side, but still can improve the UX compared to what we have right now.

Maybe we could improve it at a later point that users can configure whatever coin they want to have in this tab section.

WDYT?

**Reply by chimp1984** [reply]
I think it would remove one of the goals of that proposal to make XMR more visible for newbies. XMR is 90% of trade volume so it clearly deserves a front seat IMO.

**Reply by m52go** [reply]
Lately it has been more in the 50% range [0]. I think the 90% is mostly from those brief periods of times whales came and trade 1-2 BTC at a time. With regards to the UX change, the vast majority of XMR traders who use Bisq already know that it can trade XMR.

I talked to UX-P to try understanding their motivations and this whole thing seems like a solution in search of a problem based on unclear/questionable premises.

[0]

![Screenshot from 2022-03-16 12-32-08](https://user-images.githubusercontent.com/735155/158644208-fecd57e0-e068-4f5b-8bdc-44f6924d28d3.png)

**Reply by UX-P** [reply]
This data reinforces the fact that the BTC-Altcoin market on Bisq could benefit from a little more exposure on the UI. What is your feedback regarding the navigation design below? 

Buy/Sell > BTC/BSQ/Altcoins

**Reply by m52go** [reply]
I think that's great. It would be great to get a spotlight on BSQ. If BSQ got its own tab, there could even be a line or two about what BSQ is and why a user might want to get some.

**Comment by UX-P** [comment]
> I think the two currency dropdowns on the main screen are confusing

In regards to the 'Market' tab I agree, and it is actually best to keep the landing page at its current state for now and prioritise the menu navigation, so we can leave it as is. 

> Having Buy > BTC | BSQ | Other

Without XMR or another altcoin on display, the design does not meet the objective of exposing the options available on Bisq and the term ‘other’ becomes ambiguous and less understandable.

Alternatively, we could do: BTC/BSQ/Altcoins

The term Altcoins is currently being used on Bisq so shouldn’t be unfamiliar or disturbing to the current Bisq user, it describes all other coins that aren’t BTC and it increases the visibility of all coins available on Bisq to trade using BTC, making this an easier journey for the new user.

The only potential confusion would be that the payment method is also labelled as ‘Altcoins’ but shouldn’t be too much of an issue because as @chimp1984 mentioned, this method is used for all Altcoins and only XMR is associated with 'Altcoin Instant' therefore, the design could be simplified to this:

<img width="635" alt="Screenshot 2022-03-16 at 16 25 59" src="https://user-images.githubusercontent.com/92308399/158640103-ec42f314-aa49-4235-b181-6de09041ed74.png">

XMR will still require both:

<img width="634" alt="Screenshot 2022-03-16 at 16 42 00" src="https://user-images.githubusercontent.com/92308399/158642600-ca082e25-b2b4-491d-bcf8-24c7893c936e.png">

I suggest aiming to get in an initial navigation design, set KPI’s measuring trading volume, error rate etc and we can report on any significant changes/feedback after a certain timeframe e.g. 3 months, then discuss the best course of action for any necessary improvements.

**Reply by chimp1984** [reply]
> Without XMR or another altcoin on display, the design does not meet the objective of exposing the options available on Bisq and the term ‘other’ becomes ambiguous and less understandable.

Agree. 

> Alternatively, we could do: BTC/BSQ/Altcoins

USDT might become more relevant at some point and Altcoin is not covering it well IMO.

> XMR will still require both:
We could use a toggle button instead and use the XMR-autoconfirm wording there.

**Reply by ripcurlx** [reply]
> USDT might become more relevant at some point and Altcoin is not covering it well IMO.

We could add another tab with "Stablecoins" ?

**Reply by UX-P** [reply]
The issue with this could be that Stablecoin may not be familiar enough as a general term to be exposed on the 2nd level and could add confusion to the user journey particularly next to ‘Altcoins’: 

Altcoins/Stablecoins

I suggest, if more than 3 coins do not fit under the category ‘Altcoins’ then the additional ‘Other’ tab becomes relevant again but this time meaning any coin other than altcoin, like this: 

BTC/BSQ/Altcoins/Other

If the ‘Other’ tab only includes stablecoins, the filter can be labelled as ‘Choose Stablecoin’ so that the term is slowly introduced to bisq.

Does this seem like a reasonable alternative?

**Reply by chimp1984** [reply]
> BTC/BSQ/Altcoins/Other

I think we really should stick to the 3 most relevant and keep all the rest under "Other"
BTC/BSQ/XMR/Other

**Reply by ripcurlx** [reply]
I quickly took the time to implement parts of it into the current UI to see more easily what else we need to tackle with this change.

![Bildschirmfoto 2022-03-22 um 12 33 35](https://user-images.githubusercontent.com/170962/159472996-b8258edc-bf35-4259-a6c1-46e081a7a554.png)
![Bildschirmfoto 2022-03-22 um 12 33 45](https://user-images.githubusercontent.com/170962/159472997-7f4de59c-d362-4fe1-90cf-0e2ab11dc7b6.png)
![Bildschirmfoto 2022-03-22 um 12 33 50](https://user-images.githubusercontent.com/170962/159473001-5030b73e-a643-4fc2-b116-d58431bf12b2.png)

I personally have the feeling it would be nice if we give the user more confidence what they are doing when clicking on take offer and create offer by replacing the `Select trading account` on top which is actually a duplication anyway, but instead use something like following instead or something similar.

![Bildschirmfoto 2022-03-22 um 12 41 31](https://user-images.githubusercontent.com/170962/159474330-d6b725fe-5737-43d2-873a-39da04087755.png)
![Bildschirmfoto 2022-03-22 um 12 43 21](https://user-images.githubusercontent.com/170962/159474343-b5762708-d041-41a1-b8b7-8204ba76f22b.png)

WDYT?

**Reply by chimp1984** [reply]
Looks good. I don't understand the last part.
To have the "create offer" and "take offer" in the same level might be a bit confusing. To add another tab layer for it might be an option, but also not great. To show it in a popup instead as tab would be another option but likely more efford and we would get stacking popups then later... 
To leave it that way but add some separator or make crete/take offer graphically a bit different, might do the job as well. I think it should just be separated from the other tab buttons as it has another context.

**Reply by UX-P** [reply]
> I quickly took the time to implement parts of it into the current UI to see more easily what else we need to tackle with this change.

Thanks! Good point about the offer tabs

> I think it should just be separated from the other tab buttons as it has another context.

'Create Offer' and 'Take Offer' could be used as a header and removed from the tab. The 'x' is not necessary as there is a cancel button already available on the bottom left of these pages.

Does this work? When the user clicks 'create offer' (this example only shows the top section of the create offer page):

Buy> BTC > Create Offer
<img width="637" alt="Screenshot 2022-03-22 at 23 15 22" src="https://user-images.githubusercontent.com/92308399/159591759-8687ec63-40d6-4304-8153-b71e92c0e1d6.png">
> I think we really should stick to the 3 most relevant and keep all the rest under "Other"
BTC/BSQ/XMR/Other

Ok, I am currently setting up a usability test to collect feedback from Bisq's current and potential users to ensure the navigational change is understandable and not disruptive. I'll use a hi-fi prototype and summarise the feedback. I should be ready to share my findings in a week, so we can then start on the implementation.

This sort of qualitative information should be useful for other UI redesigns in the future.

**Reply by ripcurlx** [reply]
> 'Create Offer' and 'Take Offer' could be used as a header and removed from the tab. The 'x' is not necessary as there is a cancel button already available on the bottom left of these pages.
> 
> Does this work? When the user clicks 'create offer' (this example only shows the top section of the create offer page):

I like that solution as it makes it clearer what is going on by keeping the navigational structure (Buy > BTC > Create/Take offer)

**Reply by ripcurlx** [reply]
Implementation wise I can give it a quick shot today if I find any obstacles doing it like this.

**Reply by ripcurlx** [reply]
Unfortunately took a while until I was able to work again on the new navigation. Finally I think I have everything covered except the adaptions for take and create offer.

**Old Navigation Fiat**
<img width="1312" alt="Bildschirmfoto 2022-04-01 um 13 49 05" src="https://user-images.githubusercontent.com/170962/161258323-5f609cd7-b116-4cef-b95a-ac4744c20459.png">
**New Navigation Fiat**
<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 49 08" src="https://user-images.githubusercontent.com/170962/161258341-0f38cba0-5121-49d6-914a-6b16a49bc401.png">
**Old Navigation Altcoin**
<img width="1312" alt="Bildschirmfoto 2022-04-01 um 13 49 17" src="https://user-images.githubusercontent.com/170962/161258346-f073a850-0ce9-495a-bf8f-b55f035bcbe8.png">
**New Navigation Altcoin**
<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 49 26" src="https://user-images.githubusercontent.com/170962/161258349-57a3cd0d-c4b7-447e-83bf-d67a4a60ab47.png">
**Old Navigation Buy**
<img width="1312" alt="Bildschirmfoto 2022-04-01 um 13 49 42" src="https://user-images.githubusercontent.com/170962/161258353-46654329-009c-42ff-b156-82eee9cc8a70.png">
**New Navigation Buy**
<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 49 45" src="https://user-images.githubusercontent.com/170962/161258358-98261aaf-a344-489a-8910-2d08aeee9e04.png">
<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 49 56" src="https://user-images.githubusercontent.com/170962/161258362-b9175ee5-6bb7-4d39-b60e-f48e75ae671d.png">
Payment Accounts are filtered by currency (only supported ones are shown)
<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 50 58" src="https://user-images.githubusercontent.com/170962/161258367-355d8cb0-a245-439b-8780-6e7da8046bbc.png">
<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 51 17" src="https://user-images.githubusercontent.com/170962/161258371-1664e1d3-096e-498d-b644-ed1cf2e6511f.png">
<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 51 26" src="https://user-images.githubusercontent.com/170962/161258376-7dc2daf6-e0b2-430d-b012-f07017b7eb8f.png">
<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 51 32" src="https://user-images.githubusercontent.com/170962/161258379-2c047584-b8b0-46e9-a3a1-da009525cac4.png">
<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 51 47" src="https://user-images.githubusercontent.com/170962/161258386-5398ac9f-fc8a-4289-bf6a-fda70a36d50b.png">
<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 51 53" src="https://user-images.githubusercontent.com/170962/161258393-377f9153-9b21-401e-9360-0580ae82f963.png">
**Old Sell Screen**
<img width="1312" alt="Bildschirmfoto 2022-04-01 um 13 59 21" src="https://user-images.githubusercontent.com/170962/161259267-f73bae9b-698d-4cc1-8fb2-2c26ecb08343.png">
**New Sell Screens**
<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 59 26" src="https://user-images.githubusercontent.com/170962/161259307-863f494c-d56a-450a-bc7b-1403764e0d74.png">

<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 52 49" src="https://user-images.githubusercontent.com/170962/161258400-af91137b-689f-4fc4-a708-af974c92fb8e.png">
<img width="1314" alt="Bildschirmfoto 2022-04-01 um 13 52 54" src="https://user-images.githubusercontent.com/170962/161258402-fdcf34f4-89b4-44d1-9be3-efe081ae6298.png">

**Reply by chimp1984** [reply]
What about using "Bitcoin" and "Monero" instead of BTC and XMR in the sub tab menue? I think it would make it a bit less technical.

**Reply by UX-P** [reply]
@chimp1984 I'm not sure if that works very well, BSQ cannot be changed to text and so it will appear inconsistent with the other options.
The abbreviated version promotes a more minimal design and are terms that new traders should be encouraged to recognise while trading, particularly BTC. Maybe only having Bitcoin as text on the second level 'Bitcoin/BSQ/XMR/Other, but in the title still have it read 'Sell XMR for BTC'.

**Reply by chimp1984** [reply]
Yes, true....

**Comment by ghost** [comment]
Should CLI also be updated? When I type:
```
./bisq-cli getoffers --direction=buy --currency-code=bsq
```
I'm getting `Sell BSQ (Buy BTC)` offers - to be consistent with UI I should probably get `Buy BSQ (Sell BTC)` offers using this command. 

CC: @ghubstan

**Reply by ghubstan** [reply]
`--direction=buy` ==> `(Buy BTC)`

For  fiat/btc pairs, all offers are either buy btc or sell btc.  The same applies to the bsq/btc pair.  A buy offer is to buy btc with bsq, a sell offer is sell btc for bsq.  It is less confusing if you keep in mind that any offer object's direction (buy/sell) always applies to btc.

**Reply by ghost** [reply]
Yes I understand it, however this nomenclature has been changed on UI - now you can buy BSQ, sell XMR etc. I don't have strong opinion whether it should be changed on CLI as well - just wanted to point it out.

**Reply by ghubstan** [reply]
I can change it if there is a consensus about to do so.  And it will be a lot of work on API server, client, tests, and documentation.  (A Big Project.)    I try to explain `offer.direction` in the [API ref](https://bisq-network.github.io/slate/#rpc-method-getoffers), in [API test cases](https://github.com/bisq-network/bisq/blob/ec00c9962f238d71fc0c58210c7109b97599653a/apitest/src/test/java/bisq/apitest/method/offer/CreateBSQOffersTest.java#L57) , and  [example bots](https://github.com/bisq-network/bisq-api-reference/blob/95575e4a85506d4c93ce35e7589e31b0698979bf/java-examples/src/main/java/bisq/bots/TakeBestPricedOfferToBuyBsq.java#L147).

**Reply by ghost** [reply]
"A Big Project" :  :fearful: Not worth changing all the API upside down which is a lot of work.  Ppl developing to the API can understand that everything is in BTC perspective.
